quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::time::Timestamp;
+use core::{cmp::max, time::Duration};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+/// Bandwidth-related data tracked for each sent packet
+pub struct PacketInfo {
+    /// `Estimator.delivered_bytes` at the time this packet was sent.
+    pub delivered_bytes: u64,
+    /// `Estimator.delivered_time` at the time this packet was sent.
+    pub delivered_time: Timestamp,
+    /// `Estimator.lost_bytes` at the time this packet was sent.
+    pub lost_bytes: u64,
+    /// `Estimator.first_sent_time` at the time this packet was sent.
+    pub first_sent_time: Timestamp,
+    /// The volume of data that was estimated to be in flight at the time of the transmission of this packet
+    pub bytes_in_flight: u32,
+    /// Whether the path send rate was limited by the application rather than congestion control at the time
+    /// this packet was sent
+    pub is_app_limited: bool,
```

Comment:
nit: the first line of these comments seems pretty long, can you break it into two. actually we should add a clippy check for this if possible. the current length of the comment is 108 char

Reply:
I think it would be a rustfmt check, but this check is not currently stable: https://rust-lang.github.io/rustfmt/?version=master&search=#wrap_comments